base/src/dispatcher/dispatcher.ts

```diff
@@ -579,23 +578,29 @@ export class Dispatcher implements ZLUX.Dispatcher {
         }
         break;
       case ActionType.Minimize:
-          this.log.debug("invoke Launch, which means do nothing if wrapper found: "+wrapper);
-          this.windowManager.minimize(targetId);
-          console.log(targetId);
+          if (targetId) {
+             this.log.info("the targetID is " + targetId);
+             this.windowManager.minimize(targetId);
+
+          }
           break;
       case ActionType.Maximize:
-         this.log.debug("invoke Launch, which means do nothing if wrapper found: "+wrapper);
-         this.windowManager.maximize(targetId);
-        this.log.warn('Max/Min not supported at this time. Concern if apps should be able to control other apps visibility.');
+          if (targetId) {
+             this.windowManager.maximize(targetId);
+          }
           break;
       default:
         this.log.warn("Unhandled action type = "+action.type);
       };
     });
   }
 
-  setWindowManager(windowManager:any) {
-    this.windowManager = windowManager;
+  attachWindowManager(windowManager:any): boolean{
+    if (!this.windowManager) {
+       this.windowManager = windowManager;
+       return true;
+    }
+    return false;
```

Comment:
I think in this implicit else{} case you should also do a `this.log.warn` because we would want in the logs to know that something weird was happening.